File: cpp/1128_等价多米诺骨牌对的数量_easy.cpp
class Solution {
public:
    int numEquivDominoPairs(vector<vector<int>>& dominoes) {
        int nums[100] = {0};
        int res = 0;
        for(auto dom : dominoes){
            int x = dom[0], y = dom[1];
            int id = x <= y ? 10*x+y : 10*y+x;
            res += nums[id];
            nums[id]++;
        }
        return res;
    }
};

File: python/1128_等价多米诺骨牌对的数量_easy.py
class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        nums = [0]*100
        res = 0
        for (x, y) in dominoes:
            num = (10*x+y if x<=y else 10*y+x)
            res += nums[num]
            nums[num] += 1
        return res